### ISSUE TITLE
[apex] ApexCRUDViolation not reported when return value of DML statement is used 

### ISSUE THREAD

**Issue by yarhamjohn** [pmd/pmd#4819]
**Affects PMD Version:** 
7.0.0-rc3

**Rule:**
ApexCRUDViolation - https://pmd.github.io/pmd/pmd_rules_apex_security.html#apexcrudviolation

**Description:**

**Code Sample demonstrating the issue:**

In `myMethod`, the update statement is correctly highlighted by PMD.
```java
public void myMethod()
{
    List<Account> accounts = new List<Account> {new Account(Name = 'Account Name')};
    Database.update(accounts);
}
```
![image](https://github.com/pmd/pmd/assets/7828988/f5cba11b-7db4-4cbc-8481-3bcda366a1c6)

-------------------

In `myMethod2` the update statement is incorrectly not highlighted by PMD
```java
public List<Database.SaveResult> myMethod2()
{
    List<Account> accounts = new List<Account> {new Account(Name = 'Account Name')};
    List<Database.SaveResult> results = Database.update(accounts);
    return results;
}
```
![image](https://github.com/pmd/pmd/assets/7828988/669f5fb9-53bf-402d-a62d-f94a2e728c46)

**Expected outcome:**

PMD should report a violation in `myMethod2` but doesn't. This is a false-negative.

**Running PMD through:** VSCode using https://marketplace.visualstudio.com/items?itemName=chuckjonas.apex-pmd (v0.6.2)
